impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js

```diff
@@ -328,50 +343,102 @@ define([
       });
     },
 
-    _onSelect: function(params) {
-      params.cancelBubble = true;
+    _onSelect: function(echartsEvent) {
       var model = this.model;
-      if(model.isDirty) {
-        return;
-      }
-
-      var filter = null;
-      var srcEvent = params.event.event;
-
-      var filter = this._createFilter(params);
-
+      var filter = this._createFilter(echartsEvent);
       if(filter === null) {
         return;
       }
 
       model.select({
         dataFilter: filter,
-        selectionMode: srcEvent.ctrlKey || srcEvent.metaKey ? "toggle" : "replace"
+        selectionMode: "toggle"
       });
     },
 
-    _createFilter: function(params) {
+    _createFilter: function(echartsEvent) {
       var model = this.model;
 
-      var varsMap = this._getVars(params.data.visualKey);
-
-      var keyDataCellsMap = sceneUtil.invertVars(varsMap, model);
-
+      var varMap = this._getVars(echartsEvent.data.visualKey);
+      var keyDataCellsMap = sceneUtil.invertVars(varMap, model);
       return dataUtil.createFilterFromCellsMap(keyDataCellsMap, model.data);
     },
 
-    _getVars: function(eventData) {
-      if(eventData == null) {
+    _getVars: function(visualKey) {
+      if(visualKey == null) {
         return null;
       }
 
-      var varsMap = Object.create(null);
-      var value = eventData.split(this.groupedLabelSeparator);
+      var varMap = Object.create(null);
+      var names = visualKey.split(this.groupedLabelSeparator);
       var key = "rows";
+      varMap[key] = names;
 
-      varsMap[key] = value;
+      return varMap;
+    },
+
+     /** @override */
+    _updateSelection: function() {
+      var myChart = this._chart;
+      var chartSeries = myChart.getOption().series[0].data;
+      var selectionFilter = this.model.selectionFilter;
+      var globalSelectionState = null;
+      var modelData = this.model.data;
+      if(this.colorArray.length < 1) {
+        this.colorArray = this._echartOptions.color;
+        this.colorArrayLen = this.colorArray.length;
```

Comment:
Can't these two fields be local variables instead?
Is this some kind of caching?

Reply:
There are 36 colors in `this._echartOptions.color`. So, instead of fetching it everytime a selection is made, I felt caching it would be better 😅 .

Reply:
The array is not being copied each time!
Just do and use the local variables:
```js
var colorArray = this._echartOptions.color;
var colorArrayLen = colorArray.length;
```